tests/collector/group/group.cpp

```diff
@@ -0,0 +1,170 @@
+    EXPECT_EQ(0, g.get_update_time());
+    EXPECT_FALSE(g.has_active_job());
+
+    const Group::Metadata & md = g.get_metadata();
+    EXPECT_EQ(0, md.version);
+    EXPECT_EQ(0, md.frozen);
+    EXPECT_TRUE(md.couple.empty());
+    EXPECT_TRUE(md.namespace_name.empty());
+    EXPECT_TRUE(md.type.empty());
+    EXPECT_EQ(0, md.service.migrating);
+    EXPECT_TRUE(md.service.job_id.empty());
+
+    EXPECT_FALSE(g.metadata_parsed());
+    EXPECT_EQ(0, g.get_metadata_parse_duration());
+
+    EXPECT_EQ(Group::DATA, g.get_type());
+    EXPECT_EQ(Group::INIT, g.get_status());
+
+    EXPECT_FALSE(g.have_metadata_conflict(g));
+}
+
+TEST(Group, ParseMetadataV1)
+{
+    // This test checks metadata parsing for version 1 metadata
+    // which is msgpack:ed array of groups in couple.
+
+    // Prepare metadata.
+
+    msgpack::sbuffer buffer;
+    msgpack::packer<msgpack::sbuffer> packer(&buffer);
+    packer.pack_array(3);
+    packer.pack(17);
+    packer.pack(19);
+    packer.pack(23);
+
+    // Create group and parse metadata.
+
+    Group g(17);
+    g.save_metadata(buffer.data(), buffer.size(), ::time(nullptr)*1000000000ULL);
+    ASSERT_EQ(0, g.parse_metadata());
+    EXPECT_TRUE(g.metadata_parsed());
+
+    // Check metadata values.
+
+    const Group::Metadata & md = g.get_metadata();
+    EXPECT_EQ(1, md.version);
+    EXPECT_FALSE(md.frozen);
+    EXPECT_EQ(std::vector<int>({17, 19, 23}), md.couple);
+    EXPECT_EQ("default", md.namespace_name);
+    EXPECT_TRUE(md.type.empty());
+    EXPECT_EQ(0, md.service.migrating);
+    EXPECT_TRUE(md.service.job_id.empty());
+
+    // Check if type and status weren't affected.
+
+    g.calculate_type();
+
+    EXPECT_EQ(Group::DATA, g.get_type());
+    EXPECT_EQ(Group::INIT, g.get_status());
+
+    // Simple sanity check.
+
+    EXPECT_FALSE(g.have_metadata_conflict(g));
+}
+
+TEST(Group, ParseMetadataV2)
+{
+    // This test checks metadata parsing for version 2
+    // metadata which is msgpack:ed map.
+
+    // Prepare metadata.
+
+    msgpack::sbuffer buffer;
+    msgpack::packer<msgpack::sbuffer> packer(&buffer);
+    packer.pack_map(6);
+
+    packer.pack(std::string("version"));
+    packer.pack(2);
+
+    packer.pack(std::string("frozen"));
+    packer.pack(true);
+
+    packer.pack(std::string("couple"));
+    packer.pack_array(3);
+    packer.pack(29);
+    packer.pack(31);
+    packer.pack(37);
+
+    packer.pack(std::string("namespace"));
+    packer.pack(std::string("storage"));
+
+    packer.pack(std::string("type"));
+    packer.pack(std::string("cache"));
+
+    packer.pack(std::string("service"));
+    packer.pack_map(2);
+    packer.pack(std::string("status"));
+    packer.pack(std::string("MIGRATING"));
+    packer.pack(std::string(std::string("job_id")));
+    packer.pack(std::string("12345"));
+
+    // Create group and parse metadata.
+
+    Group g(29);
+    g.save_metadata(buffer.data(), buffer.size(), ::time(nullptr)*1000000000ULL);
+    ASSERT_EQ(0, g.parse_metadata());
```

Comment:
Changed to `bool`.
https://github.com/v-nikulichev/mastermind/commit/2a852a38bbc9090a134dd23a4508b10ab85bd2dd
https://github.com/v-nikulichev/mastermind/commit/fbb4eaa191d8e21ad504147b7a24d6f0fae2d10b